src/intelligence_layer/core/model.py

```diff
@@ -442,64 +444,6 @@ def __init__(
     def eot_token(self) -> str:
         return "<|endoftext|>"
 
-    def to_instruct_prompt(
-        self,
-        instruction: str,
-        input: Optional[str] = None,
-        response_prefix: Optional[str] = None,
-    ) -> RichPrompt:
-        return self.INSTRUCTION_PROMPT_TEMPLATE.to_rich_prompt(
-            instruction=instruction, input=input, response_prefix=response_prefix
-        )
-
-
-class Llama2InstructModel(ControlModel):
-    """A llama-2-*-chat model, prompt-optimized for single-turn instructions.
-
-    If possible, we recommend using `Llama3InstructModel` instead.
```

Comment:
Im fine with removing llama2, but we should mention this in the changelog as a breaking change. If people update IL for new features, they might as well update their model OR just copy this class. Might not be the nicest way since we didn't deprecated it ourselves, but I think this is fine, there are drop-in replacements for this.

Reply:
Hmm, maybe let's keep the class in then

Reply:
And just raise a Deprecation warning